feat(traverse): add `generate_binding` and `generate_binding_current_scope` APIs in context

Dunqing opened the pull request:
These two APIs are used to create a symbol with the provided symbol name, The difference from `generate_uid` is that we don't need to create a unique name for the symbol.


If you have a better method name for this, Feel free to edit this PR directly

Dunqing commented:
* **#6806** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6806?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#6805** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6805?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
I think it'd be preferable if `generate_binding` took an `Atom` instead of a `CompactStr`. If it's an existing var name (which it must be, otherwise we'd be using `generate_uid`), an `Atom` will already exist in the arena for this symbol name, so we'd be better off reusing it. I'll submit that change in a follow-on PR.

This business with juggling `CompactStr`s and `Atom`s is a real pain!

Dunqing committed:
feat(traverse): add `generate_binding` and `generate_binding_current_scope` APIs in context (#6805)

These two APIs are used to create a symbol with the provided symbol name, The difference from `generate_uid` is that we don't need to create a unique name for the symbol.

If you have a better method name for this, Feel free to edit this PR directly
